DuckDuckGo/NetworkProtectionStatusViewModel.swift

```diff
@@ -94,12 +95,43 @@ final class NetworkProtectionStatusViewModel: ObservableObject {
         return formatter
     }()
 
+    private let featureFlagger = AppDependencyProvider.shared.featureFlagger
     private let tunnelController: (TunnelController & TunnelSessionProvider)
     private let statusObserver: ConnectionStatusObserver
     private let serverInfoObserver: ConnectionServerInfoObserver
     private let errorObserver: ConnectionErrorObserver
     private var cancellables: Set<AnyCancellable> = []
 
+    // MARK: - Tips
+
+    var canShowTips: Bool {
+        featureFlagger.isFeatureOn(.networkProtectionUserTips)
+    }
+
+    @Published
+    var showAddWidgetEducationView: Bool = false
+
+    let vpnEnabledTips: TipGrouping = {
+        // This is temporarily disabled until Xcode 16 is available.
+        // Ref: https://app.asana.com/0/414235014887631/1208528787265444/f
+        //
+        // if #available(iOS 18.0, *) {
+        //     return TipGroup(.ordered) {
+        //     VPNChangeLocationTip()
+        //     VPNUseSnoozeTip()
+        //     VPNAddWidgetTip()
+        // }
+        if #available(iOS 17.0, *) {
+            return LegacyTipGroup(.ordered) {
+                VPNChangeLocationTip()
+                VPNUseSnoozeTip()
+                VPNAddWidgetTip()
+            }
+        } else {
+            return EmptyTipGroup()
+        }
+    }()
+
```

Comment:
This is great because we'll be using the default thing in iOS 18+, my backport in iOS 17 and a glorified no-op in lower OS versions.

The iOS 18 code is commented for now as Xcode 15.4 fails to build with it.  I've created [a follow-up task](https://app.asana.com/0/414235014887631/1208528787265444/f) to re-enable.